### PR TITLE
chore(docs): document training options and RuntimePatch API usage

### LINKED ISSUE #388: Document how to use `options` with TrainerClient
We should update documentation to explain how to use options while creating TrainJob to configure RuntimePatch API.

https://sdk.kubeflow.org/en/latest/

cc @kubeflow/kubeflow-sdk-team @Fiona-Waters 

/good-first-issue
/area docs

### PR DESCRIPTION
**Pull request by prakharsingh-74**
### Description
This PR documents how to customize `TrainJob` creation using training options and the Kubernetes `RuntimePatch` API.

### Changes
* **Quickstart Guide** (`docs/source/getting-started/quickstart.rst`):
  * Added a standalone **Optional: Configure TrainJob options** section right after Step 3.
  * Demonstrated standard metadata options (`Name`, `Labels`, `Annotations`) while keeping the `train_mnist` function argument fully consistent.
  * Linked to the advanced options guide for `RuntimePatch`.
* **Trainer Options Guide** (`docs/source/train/options.rst`):
  * Revamped the document to provide a comprehensive explanation of training options.
  * Detailed the nested structure/hierarchy of the `RuntimePatch` schema (e.g. `JobSetTemplatePatch` -> `PodSpecPatch` -> `ContainerPatch`).
  * Provided a complete, copy-pasteable example of using `RuntimePatch` to mount PVC volumes, override container environment variables, set node selectors, tolerations, and container security contexts.
  * Added a self-contained dummy `train_fn` definition in the advanced code block to prevent reference mismatch errors.
* **Trainer Index** (`docs/source/train/index.rst`):
  * Added a new card **Training Options** under the Guides grid for better discoverability of advanced configuration options.

### Verification
* git diff --check
* lightweight assertion script to confirm the new section and RuntimePatch example are present

Fixes #388 

**Checklist:**

- [x] [Docs](https://www.kubeflow.org/docs/) included if any changes are user facing

### CODE REVIEW

**Comment by prakharsingh-74** [comment]
@Fiona-Waters and @andreyvelich Opening the PR for the issue #388 as the previous contributor remains inactive from the month of march and give no update on the PR

**Review by andreyvelich — COMMENTED** [review]
/ok-to-test
/retest

File: docs/source/train/options.rst [hunk 1]
```diff
@@ -1,5 +1,167 @@
...
+                  * ``metadata`` (dict)
+                  * ``spec`` (JobSpecPatch)
+                     * ``template`` (PodTemplatePatch)
+                        * ``metadata`` (dict)
+                        * ``spec`` (PodSpecPatch)
+                           * ``node_selector`` (dict)
+                           * ``tolerations`` (list[dict])
+                           * ``volumes`` (list[dict])
+                           * ``containers`` (list[ContainerPatch])
+                              * ``name`` (str) - Name of the container to patch (e.g., ``"pytorch"``).
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: The trainer container is named `node`, not `pytorch` (`constants.py:88`, looked up at `backends/kubernetes/utils.py:104`). `"pytorch"` is the Training Operator v1 `PyTorchJob` name.

```suggestion
                              * ``name`` (str) - Name of the container to patch. Must exist in the Runtime (e.g. ``"node"``).
```

**Review by andreyvelich — COMMENTED** [review]
Thanks for putting this together, @prakharsingh-74 - `options.rst` being a bare `autoclass` dump was a real gap, and the nested-hierarchy tree is exactly the piece that was missing.

I've left inline suggestions below. This review was done with the help of AI tools.

File: docs/source/train/options.rst [hunk 2]
```diff
@@ -1,5 +1,167 @@
...
+                              * ``env`` (list[dict])
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: Upstream forbids `env` here for the `node`/initializer containers (`TrainerV1alpha1ContainerPatch.env` description).

```suggestion
                              * ``env`` (list[dict]) - Not allowed for the ``node``, ``dataset-initializer``, or ``model-initializer`` containers; use ``CustomTrainer(env=...)`` or the Initializer API instead.
```

File: docs/source/train/options.rst [hunk 3]
```diff
@@ -1,5 +1,167 @@
...
+               * ``template`` (JobTemplatePatch)
+                  * ``metadata`` (dict)
+                  * ``spec`` (JobSpecPatch)
+                     * ``template`` (PodTemplatePatch)
+                        * ``metadata`` (dict)
+                        * ``spec`` (PodSpecPatch)
+                           * ``node_selector`` (dict)
+                           * ``tolerations`` (list[dict])
+                           * ``volumes`` (list[dict])
+                           * ``containers`` (list[ContainerPatch])
```
Comment (andreyvelich) [inline docs/source/train/options.rst:65]: `PodSpecPatch` has 10 fields (`options/kubernetes.py:109-118`); 6 are missing, which leaves the "scheduling parameters" claim above unsupported.

```suggestion
                        * ``spec`` (PodSpecPatch)
                           * ``service_account_name`` (str)
                           * ``volumes`` (list[dict])
                           * ``init_containers`` (list[ContainerPatch])
                           * ``containers`` (list[ContainerPatch])
                           * ``image_pull_secrets`` (list[dict])
                           * ``security_context`` (dict)
                           * ``node_selector`` (dict[str, str])
                           * ``affinity`` (dict)
                           * ``tolerations`` (list[dict])
                           * ``scheduling_gates`` (list[dict])
```

File: docs/source/train/options.rst [hunk 4]
```diff
@@ -1,5 +1,167 @@
...
+                                                           "name": "DATA_DIR",
+                                                           "value": "/mnt/data",
+                                                       }
+                                                   ],
+                                                   # Configure container security settings
+                                                   security_context={
+                                                       "privileged": True,
+                                                   },
+                                               )
+                                           ],
```
Comment (andreyvelich) [inline docs/source/train/options.rst:159]: Three fixes: `pytorch` is not a container in the Runtime (it is `node`); `env` is rejected by the API for the `node` container; and docs should not showcase `privileged: True`.

```suggestion
                                           containers=[
                                               ContainerPatch(
                                                   # Must match a container in the
                                                   # Runtime; "node" is the trainer.
                                                   name="node",
                                                   # Mount the volume
                                                   volume_mounts=[
                                                       {
                                                           "name": "data-volume",
                                                           "mountPath": "/mnt/data",
                                                       }
                                                   ],
                                                   # Harden the container
                                                   security_context={
                                                       "allowPrivilegeEscalation": False,
                                                       "capabilities": {"drop": ["ALL"]},
                                                   },
                                               )
                                           ],
```

File: docs/source/train/options.rst [hunk 5]
```diff
@@ -1,5 +1,167 @@
...
+               )
+           )
+       )
+   )
+
+   # Submit training job with the patch
+   client.train(
+       trainer=CustomTrainer(func=train_fn),
+       options=[patch]
+   )
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: Keeps `DATA_DIR` in the example via the supported path (`CustomTrainer.env`, `types.py:62`).

```suggestion
   # Submit training job with the patch. Env vars for the "node" container must be
   # set on the trainer, not via ContainerPatch.
   client.train(
       trainer=CustomTrainer(func=train_fn, env={"DATA_DIR": "/mnt/data"}),
       options=[patch],
   )
```

File: docs/source/train/options.rst [hunk 6]
```diff
@@ -1,5 +1,167 @@
...
+                           * ``containers`` (list[ContainerPatch])
+                              * ``name`` (str) - Name of the container to patch (e.g., ``"pytorch"``).
+                              * ``env`` (list[dict])
+                              * ``volume_mounts`` (list[dict])
+                              * ``security_context`` (dict)
+
+Advanced Patching Example
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here is a complete example of using ``RuntimePatch`` to mount a PVC to the training container, set custom environment variables, configure a node selector, and add a scheduling toleration:
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: Drop the env claim (not settable via `ContainerPatch` here) and wrap to ~100 chars.

```suggestion
Here is a complete example of using ``RuntimePatch`` to mount a PVC to the training
container, configure a node selector, add a scheduling toleration, and set a container
security context:
```

File: docs/source/train/options.rst [hunk 7]
```diff
@@ -1,5 +1,167 @@
-Options Reference
-=================
+Training Options
+================
+
+Training options allow you to customize the metadata and configuration of your training jobs. They are optional arguments that can be passed to the ``options`` parameter of ``TrainerClient.train()``.
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: Existing `docs/source/train/*.rst` wrap at <=96 chars; this PR adds lines up to 308 (also lines 9, 34, 39, 47).

```suggestion
Training options allow you to customize the metadata and configuration of your training
jobs. They are optional arguments that can be passed to the ``options`` parameter of
``TrainerClient.train()``.
```

File: docs/source/train/options.rst [hunk 8]
```diff
@@ -60,3 +222,4 @@ Options Reference
 .. autoclass:: kubeflow.trainer.options.ContainerPatch
    :members:
    :show-inheritance:
+
```
Comment (andreyvelich) [inline docs/source/train/options.rst]: The file now ends with a blank line; the `end-of-file-fixer` pre-commit hook will fail CI.

```suggestion
   :show-inheritance:
```

File: docs/source/getting-started/quickstart.rst [hunk 9]
```diff
@@ -99,6 +99,29 @@ Or check the job status:
...
+   job_name = client.train(
+       trainer=CustomTrainer(func=train_mnist),
+       options=[
+           Name("my-mnist-job"),
+           Labels({"team": "ml-platform"}),
+           Annotations({"description": "Quickstart training job"}),
+       ]
+   )
+
+For more advanced configurations, such as mounting volumes or customizing container resources via the ``RuntimePatch`` API, see the :doc:`../train/options` guide.
```
Comment (andreyvelich) [inline docs/source/getting-started/quickstart.rst]: `ContainerPatch` has only `name`/`env`/`volume_mounts`/`security_context` - no resources field. Resources go through `CustomTrainer(resources_per_node=...)`.

```suggestion
For more advanced configurations, such as mounting volumes, setting node selectors, or
patching container security contexts via the ``RuntimePatch`` API, see the
:doc:`../train/options` guide.
```

File: docs/source/getting-started/quickstart.rst [hunk 10]
```diff
@@ -99,6 +99,29 @@ Or check the job status:
...
+   # Submit the job with custom name, labels, and annotations
+   job_name = client.train(
+       trainer=CustomTrainer(func=train_mnist),
+       options=[
+           Name("my-mnist-job"),
+           Labels({"team": "ml-platform"}),
+           Annotations({"description": "Quickstart training job"}),
+       ]
+   )
+
```
Comment (andreyvelich) [inline docs/source/getting-started/quickstart.rst:105]: `Labels`/`Annotations`/`RuntimePatch` raise `ValueError` on non-Kubernetes backends, and this page documents the LocalProcess backend just below.

```suggestion
   )

.. note::

   ``Labels``, ``Annotations``, and ``RuntimePatch`` are Kubernetes-only. Only ``Name``
   works with the local backends described below.

```

File: docs/source/getting-started/quickstart.rst [hunk 11]
```diff
@@ -99,6 +99,29 @@ Or check the job status:
    job = client.get_job(job_name)
    print(f"Status: {job.status}")
 
+Optional: Configure TrainJob options
+------------------------------------
+
+To customize your training job, you can pass options when submitting it in Step 2. Options allow you to set custom names, labels, annotations, or configure the Kubernetes ``RuntimePatch`` API.
```
Comment (andreyvelich) [inline docs/source/getting-started/quickstart.rst]: This section sits between Step 3 and Step 4 but points back to Step 2, and the snippet submits a **second** job that rebinds `job_name` - so Step 4 then waits on a different job than Step 3 monitored. Please move this section to directly after Step 2.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this @prakharsingh-74!
/lgtm
/approve
